packages/core/src/metadata/ReflectMetadataProvider.ts

```diff
@@ -20,7 +20,11 @@ export class ReflectMetadataProvider extends MetadataProvider {
     // as we were mapping it to UnknownType which is a string. This is to prevent defaulting to JSON
     // column type, which can be often hard to revert and cause hard to understand issues with PKs.
     if (prop.kind === ReferenceKind.SCALAR && type === Object) {
-      type = String;
+      if (prop.default!==undefined) {
```

Comment:
`null` is also problematic + spacing
```suggestion
      if (prop.default != null) {
```

and i'd move this logic into the private method + rename it

Reply:
I want to know more about `null` being problemtic and why you used `prop.default != null`? 🤓
I thought if there's no default value, `prop` won't have `default` field and will be `undefined`.
I considered using `if(!prop.default)` instead of `undefined` or `null`

Reply:
you can get `default === null` and that needs the same treatment as `undefined`, doing `!= null` will catch both `null` and `undefined`. 

you dont want `!prop.default` as that would also catch empty string or zero, both valid default values.